src/d/actor/d_a_fire.cpp

```diff
@@ -4,111 +4,528 @@
+        /* SrcGObjAt HitMark */ 0,
+        /* SrcGObjAt Spl     */ 1,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ dCcG_TgSPrm_NoHitMark_e,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    // cM3dGCylS
+    {
+        /* Center */ 0.0f, 0.0f, 0.0f,
+        /* Radius */ 130.0f,
+        /* Height */ 170.0f,
+    }
+};
+static dCcD_SrcCyl at_cyl_src = {
+    // dCcD_SrcGObjInf
+    {
+        /* Flags             */ 0,
+        /* SrcObjAt  Type    */ AT_TYPE_UNK20000,
+        /* SrcObjAt  Atp     */ 0,
+        /* SrcObjAt  SPrm    */ cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsOther_e,
+        /* SrcObjTg  Type    */ 0,
+        /* SrcObjTg  SPrm    */ cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsOther_e,
+        /* SrcObjCo  SPrm    */ 0,
+        /* SrcGObjAt Se      */ 0,
+        /* SrcGObjAt HitMark */ 0,
+        /* SrcGObjAt Spl     */ 0,
+        /* SrcGObjAt Mtrl    */ 0,
+        /* SrcGObjAt SPrm    */ 0,
+        /* SrcGObjTg Se      */ 0,
+        /* SrcGObjTg HitMark */ 0,
+        /* SrcGObjTg Spl     */ 0,
+        /* SrcGObjTg Mtrl    */ 0,
+        /* SrcGObjTg SPrm    */ dCcG_TgSPrm_NoHitMark_e,
+        /* SrcGObjCo SPrm    */ 0,
+    },
+    // cM3dGCylS
+    {
+        /* Center */ 0.0f, 0.0f, 0.0f,
+        /* Radius */ 150.0f,
+        /* Height */ 150.0f,
+    }
+};
+
+
 /* 00000078-000000F8       .text _delete__8daFire_cFv */
 bool daFire_c::_delete() {
-    /* Nonmatching */
+    if (field_0x8BC) {
+        field_0x8BC->becomeInvalidEmitter();
+        field_0x8BC = NULL;
+    }
+    if (field_0x8C0) {
+        field_0x8C0->becomeInvalidEmitter();
+        field_0x8C0 = NULL;
+    }
+    if (field_0x8C4 != NULL) {
+        field_0x8C4->becomeInvalidEmitter();
+        field_0x8C4 = NULL;
+    }
+    return TRUE;
 }
 
 /* 000000F8-00000518       .text CreateInit__8daFire_cFv */
-void daFire_c::CreateInit() {
-    /* Nonmatching */
+BOOL daFire_c::CreateInit() {
+    mSwitchNo = daFire_prm::getSwitch(this);
+    mTrboxBit = daFire_prm::getTrboxBit(this);;
+
+    field_0x8EC = fopAcM_isSwitch(this, mSwitchNo) || dComIfGs_isTbox(mTrboxBit);
+
+    if (dComIfGs_isTbox(mTrboxBit)) {
+        return FALSE;
+    }
+
+    mType = daFire_prm::getType(this);
+    switch (mType) {
+        case 0:
+            if (mSwitchNo == 0xFF) {
+                mType = 2;
+            }
+            break;
+        case 1:
+        case 2:
+        default:
+            break;
+    }
+
+    field_0x901 = 0x1E;
+    mShape = daFire_prm::getShape(this);
+
+    switch (mShape) {
+        case 0:
+            break;
+        case 1:
+            scale.z = 0.0f;
+            break;
+        case 2:
+            // Setting seems to be in zyx order, weird.
+            scale.z = 1.25f;
+            scale.y = 1.25f;
+            scale.x = 1.25f;
```

Comment:
Reverse order suggests they assigned multiple variables in a oneliner, which the compiler puts backwards.
```suggestion
            scale.x = scale.y = scale.z = 1.25f;
```